azure/test/client.test.js

```diff
@@ -1,180 +1,292 @@
 var assert = require('assert');
 var client = require('../activity_logs_monitoring').forTests;
-var constants = client.constants
-var sinon = require('sinon')
+var constants = client.constants;
+var sinon = require('sinon');
 
+function fakeContext() {
+    // create a fake context object to pass into handleLogs
+    contextSpy = sinon.spy();
+    contextSpy.log = sinon.spy();
+    contextSpy.log.error = function(x) {}; // do nothing
+    contextSpy.log.warn = function(x) {}; // do nothing
+
+    return contextSpy;
+}
+
+var EventhubLogForwarderInstance = new client.EventhubLogForwarder(
+    fakeContext()
+);
+EventhubLogForwarderInstance.sendWithRetry = function(record) {}; // do nothing
+console.log(EventhubLogForwarderInstance.getLogFormat);
+var handleJsonLogsSpy = sinon.spy();
+var handleStringLogsSpy = sinon.spy();
+
+EventhubLogForwarderInstance.addTagsToJsonLog = handleJsonLogsSpy;
+EventhubLogForwarderInstance.addTagsToStringLog = handleStringLogsSpy;
 
 describe('Azure Log Monitoring', function() {
-  describe('#getLogFormat', function() {
-    it('should return string', function() {
-        eventHubMessages = '';
-        assert.equal(constants.STRING, client.getLogFormat(eventHubMessages));
-        eventHubMessages = 'foobar';
-        assert.equal(constants.STRING, client.getLogFormat(eventHubMessages));
-    });
-    it('should return string array', function() {
-      eventHubMessages = ['', 'foobar'];
-      assert.equal(constants.STRING_ARRAY, client.getLogFormat(eventHubMessages));
-    });
-    it('should return json object', function() {
-      eventHubMessages = {'key': 'value', 'otherkey':'othervalue'};
-      assert.equal(constants.JSON_OBJECT, client.getLogFormat(eventHubMessages));
+    describe('#getLogFormat', function() {
+        it('should return string', function() {
+            eventHubMessages = '';
+            assert.equal(
+                constants.STRING,
+                EventhubLogForwarderInstance.getLogFormat(eventHubMessages)
+            );
+            eventHubMessages = 'foobar';
+            assert.equal(
+                constants.STRING,
+                EventhubLogForwarderInstance.getLogFormat(eventHubMessages)
+            );
+        });
+        it('should return string array', function() {
+            eventHubMessages = ['', 'foobar'];
+            assert.equal(
+                constants.STRING_ARRAY,
+                EventhubLogForwarderInstance.getLogFormat(eventHubMessages)
+            );
+        });
+        it('should return json object', function() {
+            eventHubMessages = { key: 'value', otherkey: 'othervalue' };
+            assert.equal(
+                constants.JSON_OBJECT,
+                EventhubLogForwarderInstance.getLogFormat(eventHubMessages)
+            );
+        });
+        it('should return json no records', function() {
```

Comment:
The should string is confusing. Can you add a comment about how that maps to a JSON array?

Reply:
This was added before I started working on the log forwarder, but I will make it more clear.